test/cppsuite/test_harness/workload/database_model.h

```diff
@@ -49,112 +55,188 @@ struct value_t {
 };
 
 /* A collection is made of mapped Key objects. */
-struct collection_t {
-    std::map<key_value_t, key_t> keys;
-    std::map<key_value_t, value_t> values;
+class collection {
+    public:
+    collection(const uint64_t id, const uint64_t key_count, const std::string name)
+        : id(id), _key_count(key_count), name(name)
+    {
+    }
+
+    int
+    get_new_key()
+    {
+        std::lock_guard<std::mutex> lg(_mtx);
+        return _key_count++;
+    }
+
+    std::map<key_value_t, key_t>
+    get_keys(const uint64_t &id)
+    {
+        std::lock_guard<std::mutex> lg(_mtx);
+        return (keys);
+    }
+
+    value_t
+    get_record(const uint64_t &id, const char *key)
+    {
+        std::lock_guard<std::mutex> lg(_mtx);
+        return (values.at(key));
+    }
+
+    void
+    insert_record(const uint64_t &id, const char *key, const char *value)
+    {
+        std::lock_guard<std::mutex> lg(_mtx);
+        keys[key].exists = true;
+        value_t v;
+        v.value = key_value_t(value);
+        values.emplace(key_value_t(key), v);
+    }
+
+    void
+    update_record(const uint64_t &id, const char *key, const char *value)
+    {
+        std::lock_guard<std::mutex> lg(_mtx);
+        values.at(key).value = key_value_t(value);
+    }
+
+    void
+    delete_record(const uint64_t &id, const char *key)
+    {
+        std::lock_guard<std::mutex> lg(_mtx);
+        keys.at(key).exists = false;
+    }
+
+    std::map<key_value_t, key_t> keys = {};
+    std::map<key_value_t, value_t> values = {};
+    const std::string name;
+    const uint64_t id;
+
+    private:
+    std::mutex _mtx;
+    uint64_t _key_count;
 };
 
 /* Representation of the collections in memory. */
 class database {
     public:
     /*
-     * Add a new collection following the standard naming pattern. Currently this is the only way to
-     * add collections which is supported by all components.
+     * Add a new collection, this will create the underlying collection in the database.
      */
-    std::string
-    add_collection()
+    void
+    add_collection(uint64_t key_count = 0)
     {
+        WT_SESSION *session = connection_manager::instance().create_session();
         std::lock_guard<std::mutex> lg(_mtx);
         std::string collection_name = build_collection_name(_next_collection_id);
-        _collections[collection_name] = {};
+        _collections.emplace(
+          _next_collection_id, new collection(_next_collection_id, key_count, collection_name));
+        ++_next_collection_id;
+        testutil_check(session->create(session, collection_name.c_str(), DEFAULT_FRAMEWORK_SCHEMA));
```

Comment:
DEFAULT_FRAMEWORK_SCHEMA could be an input arg of this function
add_collection(uint64_t, const string& = DEFAULT_FRAMEWORK_SCHEMA)

Reply:
Or it should be an arg of the populate() method. But this will have other impacts if we give the user the possibility to give a different schema. Should be another ticket.